### PR TITLE
FIX: empty filename in CAM workbench saves to the current project folder

### PR DESCRIPTION
**Pull request by alexfacciorusso**
Previously, if the user didn't set a default gcode path, the file was saved to getcwd. Now, it attempts to take the active document's path first.

I would like suggestions about what the standard behaviour should we use, but in general, this needed to be fixed as in Windows at least, it was saving to the same `freecad.exe` folder!

### CODE REVIEW

**Comment by alexfacciorusso** [comment]
Potentially solves #8875

**Comment by maxwxyz** [comment]
@krandelbord could you confirm this as it's your issue?

**Comment by alexfacciorusso** [comment]
Additional fix: when the dialogs in CAM were cancelled, the action was totally ignored, resulting in the app still saving the file even if the user closed the save dialog or clicked on "cancel". Also refactored the CAM tests for file paths so that there's no issues in check for equality

File: src/Mod/CAM/Path/Post/Command.py [hunk 1]
```diff
@@ -157,12 +158,15 @@ def _write_file(self, filename, gcode, policy):
                     Path.Log.debug(filename)
                     with open(filename, "w") as f:
                         f.write(gcode)
+                else: return
```
Comment (alexfacciorusso) [inline src/Mod/CAM/Path/Post/Command.py:161]: Here, the dialog "cancel" action was ignored, resulting in the action being performed even if the dialog was cancelled

File: src/Mod/CAM/Path/Post/Command.py [hunk 2]
```diff
@@ -137,6 +137,7 @@ def _write_file(self, filename, gcode, policy):
                 Path.Log.debug(filename)
                 with open(filename, "w") as f:
                     f.write(gcode)
+            else: return
```
Comment (alexfacciorusso) [inline src/Mod/CAM/Path/Post/Command.py:140]: Here, the dialog "cancel" action was ignored, resulting in the action being performed even if the dialog was cancelled

**Comment by alexfacciorusso** [comment]
This is mergeable now, let me know when we can merge so that people can test this in the nightly builds

**Comment by luzpaz** [comment]
Merge meetings happen every Monday. But there is a hackathon underway, so this may get reviewed earlier than that. TIA for your contribution and effort

**Pull request merged**